src/CognitoDatasetManager.js

```diff
@@ -1,230 +1,376 @@
-import { EventEmitter } from "events";
-import Amplify, { Auth } from "aws-amplify";
+import { EventEmitter } from "events"
+import Amplify, { Auth, Storage } from "aws-amplify"
+import seamlessImmutable from "seamless-immutable"
+const { from: seamless } = seamlessImmutable
 
 class CognitoDatasetManager extends EventEmitter {
-  type = "cognito";
-
-  constructor({ authConfig, dummyUser } = {}) {
-    super();
-    if (!authConfig.Auth.region) throw new Error("Auth region is required");
+  type = "cognito"
+  ds = null
+  proxyUrl = "https://cors-anywhere.herokuapp.com/"
```

Comment:
this should eventually be able to be passed into the constructor, because enterprise clients will want to run their own proxy, but it's fine for now